runtime/doc/diagnostic.txt

```diff
@@ -0,0 +1,524 @@
+                                 diagnostic options.
+
+disable({bufnr}, {namespace})                       *vim.diagnostic.disable()*
+                Disable diagnostics in the given buffer.
+
+                Parameters: ~
+                    {bufnr}      number|nil Buffer number. Defaults to the
+                                 current buffer.
+                    {namespace}  number|nil Only disable diagnostics for the
+                                 given namespace.
+
+enable({bufnr}, {namespace})                         *vim.diagnostic.enable()*
+                Enable diagnostics in the given buffer.
+
+                Parameters: ~
+                    {bufnr}      number|nil Buffer number. Defaults to the
+                                 current buffer.
+                    {namespace}  number|nil Only enable diagnostics for the
+                                 given namespace.
+
+get({bufnr}, {namespace})                               *vim.diagnostic.get()*
+                Get all diagnostics for the buffer.
+
+                Parameters: ~
+                    {bufnr}      number|nil Buffer number to get diagnistics
+                                 from. Use 0 for current buffer. When omitted,
+                                 retrieve diagnostics from all buffers.
+                    {namespace}  number|nil Limit diagnostics to the given
+                                 namespace.
+
+                Return: ~
+                    table A list of diagnostic items |diagnostic-structure|.
+
+                                                  *vim.diagnostic.get_count()*
+get_count({bufnr}, {severity}, {namespace})
+                Get the number of diagnostics with the given severity.
+
+                Useful for showing diagnostic counts in the |status-line|.
+                Example: >
+
+                 function! DiagnosticStatus() abort
+                   let sl = ''
+                   let sl .= 'E: %{luaeval("vim.diagnostic.get_count(0, vim.diagnostic.severity.ERROR)")} '
+                   let sl .= 'W: %{luaeval("vim.diagnostic.get_count(0, vim.diagnostic.severity.WARN)")}'
+                   return sl
+                 endfunction
+<
+
+                Then simply add `%{DiagnosticStatus()}` to 'statusline'.
+
+                Parameters: ~
+                    {bufnr}      number|nil Buffer number. Use 0 for current
+                                 buffer. When omitted, count diagnostics from
+                                 all buffers.
+                    {severity}   |vim.diagnostic.severity| Count only
+                                 diagnostics for the given severity. When
+                                 omitted, count all diagnostics.
+                    {namespace}  number|nil Count only diagnostics from the
+                                 given namespace.
+
+                                       *vim.diagnostic.get_line_diagnostics()*
+get_line_diagnostics({bufnr}, {lnum}, {opts})
```

Comment:
- for parallel form with `get()` we can name this `get_by_linenr`. (Don't need to repeat "diagnostics" in the name)
- alternatively I would consider adding an optional `opt.linenr` param to `get()`, then we don't need this extra function.

```suggestion
get_by_linenr({bufnr}, {lnum}, {opts})
```

Reply:
I actually like just adding a line number to `get()`. I don't think the symmetry with `show_line_diagnostics` is important.